File: tags/symmetric-build-3.5.19+b.1/symmetric-1.4.3/src/main/java/org/jumpmind/symmetric/model/NodeGroupLink.java
/*
 * SymmetricDS is an open source database synchronization solution.
 *   
 * Copyright (C) Chris Henson <chenson42@users.sourceforge.net>
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, see
 * <http://www.gnu.org/licenses/>.
 */

package org.jumpmind.symmetric.model;

public class NodeGroupLink {

    private static final long serialVersionUID = 1L;

    private String sourceGroupId;

    private String targetGroupId;

    private DataEventAction dataEventAction = DataEventAction.WAIT_FOR_POLL;

    public DataEventAction getDataEventAction() {
        return dataEventAction;
    }

    public void setDataEventAction(DataEventAction dataEventAction) {
        this.dataEventAction = dataEventAction;
    }

    public String getSourceGroupId() {
        return sourceGroupId;
    }

    public void setSourceGroupId(String domainName) {
        this.sourceGroupId = domainName;
    }

    public String getTargetGroupId() {
        return targetGroupId;
    }

    public void setTargetGroupId(String targetDomainName) {
        this.targetGroupId = targetDomainName;
    }
}

File: symmetric-client-clib/src/model/NodeGroupLink.c
/**
 * Licensed to JumpMind Inc under one or more contributor
 * license agreements.  See the NOTICE file distributed
 * with this work for additional information regarding
 * copyright ownership.  JumpMind Inc licenses this file
 * to you under the GNU General Public License, version 3.0 (GPLv3)
 * (the "License"); you may not use this file except in compliance
 * with the License.
 *
 * You should have received a copy of the GNU General Public License,
 * version 3.0 (GPLv3) along with this library; if not, see
 * <http://www.gnu.org/licenses/>.
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
#include "model/NodeGroupLink.h"


void SymNodeGroupLink_destroy(SymNodeGroupLink *this) {
    free(this);
}

SymNodeGroupLink * SymNodeGroupLink_new(SymNodeGroupLink *this) {
    if (this == NULL) {
        this = (SymNodeGroupLink *) calloc(1, sizeof(SymNodeGroupLink));
    }
    this->destroy = (void *) &SymNodeGroupLink_destroy;
    return this;
}
